Source/DafnyTestGeneration/GenerateTestsCommand.cs

```diff
@@ -10,9 +10,16 @@ namespace Microsoft.Dafny;
 public class GenerateTestsCommand : ICommandSpec {
   public IEnumerable<Option> Options =>
     new Option[] {
+      // IMPORTANT: Before adding new options, make sure they are
```

Comment:
How would a developer know that they should do that? Why doesn't `CopyForProcedure` use the DafnyOptions clone constructor instead

Reply:
In short, merging [this one line PR to Boogie](https://github.com/boogie-org/boogie/pull/725) would allow using the DafnyOption's clone constructor and indeed make the code more robust.

The problem is that the `ProcsToCheck` field of `DafnyOptions` (inherited from Boogie's `CommadLineOptions`) does not have a setter. The clone constructor copies the field anyway using reflection but `ProcsToCheck` is a list, so the constructor copies the reference and one cannot modify the 'copied' list without modifying the original. The best solution would be to add a setter, like I do in the PR to Boogie above.